lib/src/component/abstract_transition.dart

```diff
@@ -311,9 +380,21 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps,
   // --------------------------------------------------------------------------
 
   /// Method that will be called right before the [AbstractTransitionComponent] begins to show.
+  ///
+  /// Will not work when the component is controlled using [TransitionPropsMixin.shown].
+  /// Use [TransitionPropsMixin.onWillShow] instead.
+  ///
+  /// __Deprecated.__ Use [TransitionPropsMixin.onWillShow] instead.
+  @Deprecated('Use props.onWillShow instead.')
   void prepareShow() {}
 
   /// Method that will be called right before the [AbstractTransitionComponent] begins to hide.
+  ///
+  /// Will not work when the component is controlled using [TransitionPropsMixin.shown].
+  /// Use [TransitionPropsMixin.onWillHide] instead.
+  ///
+  /// __Deprecated.__ Use [TransitionPropsMixin.onWillHide] instead.
+  @Deprecated('Use props.onWillHide instead.')
```

Comment:
Do we have a migration path for WSD components consuming these APIs? https://sourcegraph.wk-dev.wdesk.org/search?q=prepare%28Hide%7CShow%29+repo:%5Egithub%5C.com/Workiva/web_skin_dart%24+&patternType=regexp#3